testsuite/summarize_tests.py

```diff
@@ -38,7 +38,7 @@ def parse_result_file(fname):
     results = jp.JUnitXml.fromfile(fname)
 
     failed_tests = ['.'.join((case.classname, case.name)) for case in results
-                    if case.result and not isinstance(case.result, jp.junitparser.Skipped)]
+                    if case.result and not isinstance(case.result[0], jp.junitparser.Skipped)]
```

Comment:
To protect against future changes and make sure we do not underreport failures, I would suggest to add an assertion here by putting the following right ahead of this list comprehension:
```python
assert all(len(case.result) == 1 for case in results if case.result), "Case result has unexpected length > 1"
```